intel-sgx/dcap-artifact-retrieval/src/provisioning_client/intel.rs

```diff
@@ -280,12 +281,19 @@ impl<'inp> PckCrlService<'inp> for PckCrlApi {
 /// See: <https://api.portal.trustedservices.intel.com/documentation#pcs-revocation-v4>
 impl<'inp> ProvisioningServiceApi<'inp> for PckCrlApi {
     type Input = PckCrlIn;
-    type Output = PckCrl;
+    type Output = PckCrl<Unverified>;
 
     fn build_request(&self, input: &Self::Input) -> Result<(String, Vec<(String, String)>), Error> {
+        let ca = match input.ca {
+            DcapArtifactIssuer::PCKProcessorCA => "processor",
+            DcapArtifactIssuer::PCKPlatformCA => "platform",
+            DcapArtifactIssuer::SGXRootCA => {
+                return Err(Error::PCSError(StatusCode::BadRequest, "Invalid ca parameter"));
```

Comment:
`SGXRootCA` is a valid `DcapArtifactIssuer` according to https://github.com/fortanix/rust-sgx/pull/756/files#diff-9d66965f69a5767ea5d3d70d11c2cddabf121a85a2cfc2ef347119a2e3d39ed1R163, why do we return invalid CA parameter here? Should we rather return SGXRootCA not supported in this case or do we need to modify the enum values allowed in `DcapArtifactIssuer`.

Reply:
The [provisioning service](https://api.portal.trustedservices.intel.com/content/documentation.html#pcs-revocation-v4) only allows getting the CRL for the PCK platform CA and PCK Processor CA.